src/stcal/tweakreg/tweakreg.py

```diff
@@ -0,0 +1,342 @@
+                            save_abs_catalog=save_abs_catalog,
+                            output_dir=abs_catalog_output_dir)
+
+    # Check that there are enough GAIA sources for a reliable/valid fit
+    num_ref = len(ref_cat)
+    if num_ref < abs_minobj:
+        msg = (f"Not enough sources ({num_ref}) in the reference catalog "
+            "for the single-group alignment step to perform a fit. "
+            "Skipping alignment to the input reference catalog!")
+        raise TweakregError(msg)
+
+    # align images:
+    # Update to separation needed to prevent confusion of sources
+    # from overlapping images where centering is not consistent or
+    # for the possibility that errors still exist in relative overlap.
+    xyxymatch_gaia = XYXYMatch(
+        searchrad=abs_searchrad,
+        separation=abs_separation,
+        use2dhist=abs_use2dhist,
+        tolerance=abs_tolerance,
+        xoffset=0.0,
+        yoffset=0.0
+    )
+
+    # Set group_id to same value so all get fit as one observation
+    # The assigned value, 987654, has been hard-coded to make it
+    # easy to recognize when alignment to GAIA was being performed
+    # as opposed to the group_id values used for relative alignment
+    # earlier in this step.
+    for corrector in correctors:
+        corrector.meta["group_id"] = 987654
+        if ("fit_info" in corrector.meta and
+                "REFERENCE" in corrector.meta["fit_info"]["status"]):
+            del corrector.meta["fit_info"]
+
+    # Perform fit
+    try:
+        align_wcs(
+            correctors,
+            refcat=ref_cat,
+            enforce_user_order=True,
+            expand_refcat=False,
+            minobj=abs_minobj,
+            match=xyxymatch_gaia,
+            fitgeom=abs_fitgeometry,
+            nclip=abs_nclip,
+            sigma=(abs_sigma, "rmse")
+        )
+    except ValueError as e:
+        msg = e.args[0]
+        if (msg == "Too few input images (or groups of images) with "
+                "non-empty catalogs."):
+            msg += ("At least one exposure is required to align images "
+                    "to an absolute reference catalog. Alignment to an "
+                    "absolute reference catalog will not be performed.")
+            raise TweakregError(msg) from None
+        raise
+
+    except RuntimeError as e:
+        msg = e.args[0]
+        if msg.startswith("Number of output coordinates exceeded allocation"):
+            # we need at least two exposures to perform image alignment
+            msg += ("Multiple sources within specified tolerance "
+                    "matched to a single reference source. Try to "
+                    "adjust 'tolerance' and/or 'separation' parameters. "
+                    "Alignment to an absolute reference catalog will "
+                    "not be performed.")
+            raise TweakregError(msg) from None
+        raise
+
+    return correctors
+
+
+def _parse_refcat(abs_refcat: str | Path,
```

Comment:
```suggestion
def _parse_refcat(abs_refcat: str,
```
The "lower" usage below won't work for a `Path`. This was found by manually running mypy on this branch. I don't think the type hints are tested in the CI. Would you run mypy to check these?

Reply:
`lower()` should work with `Path` if it is wrapped with `str()`: `str(abs_refcat).lower()`